packs/device42/actions/device_name_list.py

```diff
@@ -0,0 +1,66 @@
+from st2actions.runners.pythonrunner import Action
+import requests
+
+
+class DeviceNameList(Action):
+    def run(self, type=None, service_level=None, in_service=None, customer=None, tags=None,
+            blade_host_name=None, virtual_host_name=None, building_id=None, building=None,
+            room_id=None, room=None, rack_id=None, rack=None, serial_no=None,
+            serial_no_contains=None, asset_no=None, name=None, tags_and=None, uuid=None,
+            is_it_switch=None, is_it_virtual_host=None, is_it_blade_host=None, hardware=None,
+            hardware_ids=None, os=None, virtual_subtype=None, last_updated_lt=None,
+            last_updated_gt=None, first_added_lt=None, first_added_gt=None,
+            custom_fields_and=None, custom_fields_or=None):
+
+        d42_server = self.config.get('d42_server', None)
+        if not d42_server:
```

Comment:
If possible, it would be great if you could re-order the arguments so the required ones are listed first and they don't default to None.

This way it's possible to use code (method) introspection to figure out which arguments are required and which not.